lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -1133,27 +1166,89 @@ static bool DeserializeAllCompilerFlags(swift::CompilerInvocation &invocation,
       /// serialized AST.
       auto deserializeCompilerFlags = [&]() -> bool {
         auto result = invocation.loadFromSerializedAST(moduleData);
-        if (result == swift::serialization::Status::Valid) {
-          if (discover_implicit_search_paths) {
-            for (auto &searchPath : searchPaths) {
-              std::string path = remap(searchPath.Path);
-              if (!searchPath.IsFramework) {
-                if (known_import_search_paths.insert(path).second)
-                  import_search_paths.push_back(path);
-              } else {
-                swift::SearchPathOptions::FrameworkSearchPath
-                    framework_search_path(path, searchPath.IsSystem);
-                if (known_framework_search_paths.insert(path).second)
-                  framework_search_paths.push_back(framework_search_path);
-              }
+        if (result != swift::serialization::Status::Valid) {
+          error << "Could not deserialize " << info.name << ":\n"
+                << getImportFailureString(result) << "\n";
+          return false;
+        }
+        if (discover_implicit_search_paths) {
+          for (auto &searchPath : searchPaths) {
+            std::string path = remap(searchPath.Path);
+            if (!searchPath.IsFramework) {
+              if (known_import_search_paths.insert(path).second)
+                import_search_paths.push_back(path);
+            } else {
+              swift::SearchPathOptions::FrameworkSearchPath
+                  framework_search_path(path, searchPath.IsSystem);
+              if (known_framework_search_paths.insert(path).second)
+                framework_search_paths.push_back(framework_search_path);
             }
           }
-          return true;
         }
+        auto exists = [&](llvm::StringRef path) {
+          if (FileSystem::Instance().Exists(path))
+            return true;
+          HEALTH_LOG_PRINTF("Ignoring missing Swift plugin at path: %s",
+                            path.str().c_str());
+          return false;
+        };
+        
+        // Discover, rewrite, and unique compiler plugin paths.
+        for (auto path : extended_validation_info.getPluginSearchPaths()) {
```

Comment:
Are we worried about reading invalid local paths in distributed swiftmodules? LLDB could read paths to local macros from a different machine/configuration, it could make LLDB load undesired macro plugins on different machines. Similar to the issue we have with the search paths pointing to `/Applications/Xcode.app/`.

If so, LLDB could apply those paths only for local non-distributed module using the `isNonUserModule` check, if a module is non-user it's likely from the SDK, so we could ignore those paths. Alternatively, if we expect macros to be in the SDK we could filter out search paths for distributed modules that point outside the SDK.

Reply:
Is `isNonUserModule` encoded in the validation info somewhere? I couldn't find it on first glance.

Reply:
It is available on `ModuleDecl` and `ModuleEntity` only at this time, but it only checks the path of the module file iirc so it should be possible to bring it down in the module loading logic as needed.

Reply:
We no longer have the original path to the .swiftmodule file when debugging from a .dSYM where all .swiftmodule files have been concatenated into the `__swift_ast` section of the symbol-rich binary.